velox/type/Conversions.h

```diff
@@ -59,11 +59,60 @@ struct Converter {
   }
 };
 
-/// Presto compatible cast of strings to boolean. There is a set of strings
-/// allowed to be casted to boolean. These strings are `t, f, 1, 0, true, false`
-/// and their upper case equivalents. Casting from other strings to boolean
-/// throws.
-Expected<bool> castToBoolean(const char* data, size_t len);
+/// This function casts a string to a boolean value. There is a set of strings
+/// allowed to be casted to boolean. Casting from other strings to boolean
+/// throws. By default, Presto compatible cast is conducted. The allowed strings
+/// are `t, f, 1, 0, true, false` and their upper case equivalents.
+/// @tparam TPolicy The specified cast policy. When set as SparkCastPolicy, the
+/// allowed strings are `t, f, 1, 0, y, n, true, false, yes, no` and their upper
+/// case equivalents.
+template <typename TPolicy>
+Expected<bool> castToBoolean(const char* data, size_t len) {
+  const auto& TU = static_cast<int (*)(int)>(std::toupper);
+
+  if (len == 1) {
+    auto character = TU(data[0]);
+    if (character == 'T' || character == '1') {
+      return true;
+    }
+    if (character == 'F' || character == '0') {
+      return false;
+    }
+    if constexpr (std::is_same_v<TPolicy, SparkCastPolicy>) {
+      if (character == 'Y') {
+        return true;
+      }
+      if (character == 'N') {
+        return false;
+      }
+    }
+  }
+
+  if ((len == 4) && (TU(data[0]) == 'T') && (TU(data[1]) == 'R') &&
```

Comment:
nit: perhaps, add a small comment to explain this code:

```
// Case-insensitive 'true'.

// Case-insensitive 'false'.

// Case-insensitive 'yes'.

// Case-insensitive 'no'.
```

Reply:
Updated. Thanks.